src/arviz_plots/plots/autocorrplot.py

```diff
@@ -0,0 +1,254 @@
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+
+    # Default max lag to 100 or max length of chain
+    if max_lag is None:
+        max_lag = 100
+
+    distribution = process_group_variables_coords(
+        dt, group=group, var_names=var_names, filter_vars=filter_vars, coords=coords
+    )
+
+    # Convert xarray.Dataset to numpy array
+    distribution_array = distribution.to_array().values
+
+    # Calculate lags up to max_lag
+    lags = np.arange(max_lag)
+
+    # Calculate autocorrelation from arviz_stats autocorr computation
+    core_base = _CoreBase()
+    acf_data = core_base.autocorr(distribution_array, axis=-1)[..., :max_lag]
+
+    dims = list(distribution.dims) + ["lag"]
+
+    # Ensure correct dimension sizes
+    coords = {}
+    for dim in distribution.dims:
+        expected_size = acf_data.shape[dims.index(dim)]
+        if dim in distribution.coords and len(distribution.coords[dim]) == expected_size:
+            coords[dim] = distribution.coords[dim]
+        else:
+            coords[dim] = np.arange(expected_size)
+
+    # Add lag coordinate
+    coords["lag"] = np.arange(acf_data.shape[-1])
+
+    # Adjust the 'lag' coordinate to match the size of the 'lag' dimension in acf_data
+    lag_size = acf_data.shape[-1]  # Size of the 'lag' dimension in acf_data
+    coords["lag"] = np.arange(lag_size)  # Update 'lag' coordinate to match the size
+
+    # Handle the case where the dimension sizes have changed
+    for dim in distribution.dims:
+        if dim in coords and acf_data.shape[dims.index(dim)] != len(coords[dim]):
+            # Adjust the dimension coordinate to match the size of the dimension in acf_data
+            coords[dim] = np.arange(acf_data.shape[dims.index(dim)])
+
+    acf_data = xr.DataArray(
+        acf_data,
+        dims=dims,
+        coords=coords,
+    )
+
+    if backend is None:
+        if plot_collection is None:
+            backend = rcParams["plot.backend"]
+        else:
+            backend = plot_collection.backend
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+
+    if plot_collection is None:
+        # Convert DataArray to Dataset for compatibility with process_facet_dims
+        pc_data = acf_data.to_dataset(name="autocorr")
+        if "column" not in pc_data.dims:
+            pc_data = pc_data.expand_dims(column=["autocorr"])
+        print(pc_data)
+
+        # Set default columns and rows for faceting
+        pc_kwargs.setdefault("cols", ["__variable__"])
+        pc_kwargs.setdefault("rows", list(set(pc_data.dims) - {"__variable__", "lag", "chain"}))
+
+        # Calculate the number of plots
+        n_plots, plots_per_var = process_facet_dims(pc_data, pc_kwargs["cols"])
+
+        # Set up figure size
+        figsize = pc_kwargs.get("plot_grid_kws", {}).get("figsize", None)
```

Comment:
You can copy all the "if plot_collection is None" logic from other plots like plotdist